### PR TITLE
feat(helm): Add podSecurityContext at deployment spec level

### PR DESCRIPTION
**Pull request by passionne**
## :id: Reference related issue. 

Issue [#9209](https://github.com/gravitee-io/issues/issues/9209) on APIM replicated to AM.
[https://gravitee.atlassian.net/browse/DEVOPS-179](DEVOPS-179)

## :pencil2: A description of the changes proposed in the pull request

In addition of the container level, you can now define a security
context at pod level.

## :memo: Test scenarios 

Edit the `values.yml` with:

* the tag 4.0.1 of docker AM images
* the right mongo uri
* set a podSecurityContext

And then:

```
kubectl create namespace my-namespace
helm install graviteeio-am . --namespace my-namespace
kubectl --namespace my-namespace get deploy graviteeio-am-management-api -o yaml | grep -A 4 'securityContext:'
```

You should see the security context defined.

## :man: :woman:@mentions of the person or team responsible for reviewing proposed changes

@gravitee-io/devops @gravitee-io/am

### CODE REVIEW

File: helm/tests/api/deployment_test.yaml [hunk 1]
```diff
@@ -112,3 +112,19 @@ tests:
...
+          podSecurityContext:
+            fsGroup: 1001
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isKind:
+          of: Deployment
+      - equal:
+          path: spec.template.spec.securityContext.fsGroup
+          value: 1001
```
Comment (leleueri) [inline helm/tests/api/deployment_test.yaml:130]: **todo**: Can we have a test case where nothing is specified for `podSecurityContext` please?

File: helm/tests/gateway/deployment_test.yaml [hunk 2]
```diff
@@ -112,3 +112,19 @@ tests:
...
+          podSecurityContext:
+            fsGroup: 1001
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isKind:
+          of: Deployment
+      - equal:
+          path: spec.template.spec.securityContext.fsGroup
+          value: 1001
```
Comment (leleueri) [inline helm/tests/gateway/deployment_test.yaml:130]: **todo**: Can we have a test case where nothing is specified for podSecurityContext please?

File: helm/tests/ui/deployment_test.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+          podSecurityContext:
+            fsGroup: 1001
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isKind:
+          of: Deployment
+      - equal:
+          path: spec.template.spec.securityContext.fsGroup
+          value: 1001
```
Comment (leleueri) [inline helm/tests/ui/deployment_test.yaml:20]: **todo**: Can we have a test case where nothing is specified for podSecurityContext please?

**Review by leleueri — CHANGES_REQUESTED** [review]
Made change request to enrich the tests but I don't know if is possible to manage the case.

**Comment by passionne** [comment]
About the test when nothing is specified for `podSecurityContext`, I do the job in template (with the `if`) and tested it manually. However, I didn't find the trick to add a unit test for that as we use an obsolete version of `helm unittest` 0.2.x and the needed feature `notExist` is available in 0.3.x only.

**Review by leleueri — APPROVED** [review]

**Pull request merged**